tests/integration/test_charm.py

```diff
@@ -0,0 +1,191 @@
+    except Exception as ex:
+        logger.info(ex)
+    return False
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test: pytest_plugin.OpsTest):
+    """Build the charm-under-test and deploy it together with related charms.
+
+    Assert on the unit status before any relations/configurations take place.
+    """
+    # build and deploy charm from local source folder
+    charm = await ops_test.build_charm(".")
+    resources = {"sdlc-image": METADATA["resources"]["sdlc-image"]["upstream-source"]}
+    await ops_test.model.deploy(charm, resources=resources, application_name=APP_NAME)
+
+    for name in USEFUL_APPS:
+        await ops_test.model.deploy(name, channel="edge")
+
+    await ops_test.model.add_relation(LEGEND_DB_APP_NAME, MONGODB_APP_NAME)
+    for app in LEGEND_APPS_CONFIG:
+        await ops_test.model.add_relation(LEGEND_DB_APP_NAME, app)
+        await ops_test.model.add_relation(GITLAB_INTEGRATOR_APP_NAME, app)
+
+    for app in LEGEND_APPS:
+        await ops_test.model.add_relation(STUDIO_APP_NAME, app)
+
+
+@pytest.mark.abort_on_fail
+async def test_config_gitlab(ops_test: pytest_plugin.OpsTest):
+    """Set the gitlab.com's application secret ID and name.
+
+    After setting the application secret ID and name, the charms should become active.
+    """
+    # We need valid information here. These should be passed in through the env variables.
+    gitlab_integrator = ops_test.model.applications["finos-legend-gitlab-integrator-k8s"]
+    await gitlab_integrator.set_config(
+        {"gitlab-client-id": GITLAB_CLIENT_ID, "gitlab-client-secret": GITLAB_CLIENT_SECRET}
+    )
+
+    # Wait for all the charms to become Active.
+    for name in APP_LIST:
+        await ops_test.model.wait_for_idle(apps=[name], status="active", timeout=2000)
+
+    # effectively disable the update status from firing
+    await ops_test.model.set_config({"update-status-hook-interval": "60m"})
+
+
+@pytest.mark.abort_on_fail
+async def test_applications_are_up(ops_test: pytest_plugin.OpsTest):
+    """Test the direct FINOS Legend connection.
+
+    The FINOS Legend application services should be accessible through their IP, ports,
+    and routes.
+    """
+    status = await ops_test.model.get_status()
+
+    for app_name in LEGEND_APPS:
+        unit_name = "%s/0" % app_name
+        address = status["applications"][app_name]["units"][unit_name]["address"]
+
+        url = "http://%s:%s%s" % (address, APP_PORTS[app_name], APP_ROUTES[app_name])
+        can_connect = check_legend_connection(app_name, url)
+
+        assert can_connect, "Could not reach %s through its IP." % app_name
+        logger.info("Successfully reached %s through its IP.", app_name)
+
+
+@pytest.mark.abort_on_fail
+async def test_nginx_ingress_integration(ops_test: pytest_plugin.OpsTest):
+    """Test the FINOS Legend connection through NGINX Ingress.
+
+    The Legend applications have been configured with the legend-host external-hostname by default,
```

Comment:
That's not true. According to the https://github.com/finos/legend-juju-sdlc-server-operator/blob/e219c3c92fa833b42ab25cc8dd7768a95b2f4d67/config.yaml#L37 , by default the SDLC charm's ``external-hostname`` is the application name. Same goes for the Engine and Studio charms.

This line needs to be corrected.

Reply:
Solved. Thank you for your feedback!